packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -76,13 +79,23 @@ int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
         // will be boxes 2x2x2 centered around a random point) will remain
         // constant as problem size is changed.
         auto const a = std::cbrt( n_values );
-        std::uniform_real_distribution<double> distribution( -a, +a );
-        std::default_random_engine generator;
-        auto random = [&distribution, &generator]() {
-            return distribution( generator );
-        };
-        for ( int i = 0; i < n; ++i )
-            random_points_host( i ) = {{random(), random(), random()}};
+        if ( point_cloud_type == "filled_box" )
```

Comment:
I think it may be cleaner to have
```c++
void generate_point_cloud(const std::string& type, ...);
```
and do the dispatch there instead of the driver.